### PR TITLE
Integrity checking

### PR DESCRIPTION
**Pull request by clyang82**
Integrity checking is to check all required conditions are met when completing the spec. we can add more. right now, add checking the components replicas meet the desired number.

Signed-off-by: clyang82 <chuyang@redhat.com>

### CODE REVIEW

File: pkg/tests/observability_alert_test.go [hunk 1]
```diff
@@ -84,10 +89,32 @@ var _ = Describe("Observability:", func() {
...
 		Expect(utils.Apply(testOptions.HubCluster.MasterURL, testOptions.KubeConfig, testOptions.HubCluster.KubeContext, yamlB)).NotTo(HaveOccurred())
 
+		By("Wait for thanos rule pods are restarted and ready")
+		// ensure the thanos rule pods are restarted successfully before processing
+		Eventually(func() error {
+			if !ThanosRuleRestarting {
+				_, newSts := utils.GetStatefulSet(testOptions, true, ThanosRuleName, MCO_NAMESPACE)
+
+				if oldSts.GetResourceVersion() == newSts.GetResourceVersion() {
+					return fmt.Errorf("The %s is not being restarted in 3 minutes", ThanosRuleName)
```
Comment (songleo) [inline pkg/tests/observability_alert_test.go:105]: Why is it `3 minutes` here?

File: pkg/tests/observability_alert_test.go [hunk 2]
```diff
@@ -130,12 +158,35 @@ var _ = Describe("Observability:", func() {
...
 
+		ThanosRuleRestarting = false
+		By("Wait for thanos rule pods are restarted and ready")
+		// ensure the thanos rule pods are restarted successfully before processing
+		Eventually(func() error {
+			if !ThanosRuleRestarting {
+				_, newSts := utils.GetStatefulSet(testOptions, true, ThanosRuleName, MCO_NAMESPACE)
+
+				if oldSts.GetResourceVersion() == newSts.GetResourceVersion() {
+					return fmt.Errorf("The %s is not being restarted in 3 minutes", ThanosRuleName)
```
Comment (songleo) [inline pkg/tests/observability_alert_test.go:177]: ditto

Comment (clyang82) [reply pkg/tests/observability_alert_test.go:177]: 👍 Need update to 10 minutes.

**Review by songleo — APPROVED** [review]

**Review by songleo — APPROVED** [review]

**Pull request merged**